codec-socks/src/main/java/io/netty/handler/codec/socksx/v4/DefaultSocks4CommandRequest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2012 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.socksx.v4;
+
+import io.netty.handler.codec.DecoderResult;
+import io.netty.util.internal.StringUtil;
+
+import java.net.IDN;
+
+/**
+ * The default {@link Socks4CommandRequest}.
+ */
+public class DefaultSocks4CommandRequest extends AbstractSocks4Message implements Socks4CommandRequest {
+
+    private final Socks4CommandType type;
+    private final String dstAddr;
+    private final int dstPort;
+    private final String userId;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param type the type of the request
+     * @param dstAddr the {@code DSTIP} field of the request
+     * @param dstPort the {@code DSTPORT} field of the request
+     */
+    public DefaultSocks4CommandRequest(Socks4CommandType type, String dstAddr, int dstPort) {
+        this(type, dstAddr, dstPort, "");
+    }
+
+    /**
+     * Creates a new instance.
+     *
+     * @param type the type of the request
+     * @param dstAddr the {@code DSTIP} field of the request
+     * @param dstPort the {@code DSTPORT} field of the request
+     * @param userId the {@code USERID} field of the request
+     */
+    public DefaultSocks4CommandRequest(Socks4CommandType type, String dstAddr, int dstPort, String userId) {
+        if (type == null) {
+            throw new NullPointerException("type");
+        }
+        if (dstAddr == null) {
+            throw new NullPointerException("dstAddr");
+        }
+        if (dstPort <= 0 || dstPort >= 65536) {
```

Comment:
either the message in the exception is wrong or the check as we also throw when port is 0